osu.Game/Online/Rooms/PlaylistExtensions.cs

```diff
@@ -39,6 +41,17 @@ public static IEnumerable<PlaylistItem> GetUpcomingItems(this IEnumerable<Playli
         }
 
         public static string GetTotalDuration(this BindableList<PlaylistItem> playlist) =>
-            playlist.Select(p => p.Beatmap.Length).Sum().Milliseconds().Humanize(minUnit: TimeUnit.Second, maxUnit: TimeUnit.Hour, precision: 2);
+            playlist.Select(p =>
+            {
+                var ruleset = p.Beatmap.Ruleset.CreateInstance();
+                double rate = 1;
+                if (p.RequiredMods.Count() > 0)
+                {
+                    List<Mod> mods = p.RequiredMods.Select(mod => mod.ToMod(ruleset)).ToList();
+                    foreach (var mod in mods.OfType<IApplicableToRate>())
+                        rate = mod.ApplyToRate(0, rate);
+                }
```

Comment:
I object to implementing this locally.

Another copy exists here:

https://github.com/ppy/osu/blob/318598730b0fea6c2fa66217a6c39b9998771486/osu.Game/Screens/Select/BeatmapInfoWedge.cs#L404-L407

If one place gets support for wind up/down (which this implementation conveniently ignores) the second will need to be updated and there is no indication that this needs to be done.

This should be a common utility helper method.

Reply:
A few other places too:

https://github.com/ppy/osu/blob/7861125e7ac1dcd48c06f5ab005870b5c111c31c/osu.Game/Beatmaps/Drawables/DifficultyIconTooltip.cs#L130-L131

https://github.com/ppy/osu/blob/10edb5461490568a06f43c443dc6cc7f19b14c8a/osu.Game/Screens/Select/Details/AdvancedStats.cs#L182-L184

https://github.com/ppy/osu/blob/310b4d90cc49fc8f26982bc0656285458d713d6c/osu.Game/Overlays/Mods/BeatmapAttributesDisplay.cs#L168-L170

I agree it's a good opportunity to combine these into a helper method. Potentially in `ModUtils`?